src/modules/tas_studio/mod.rs

```diff
@@ -2174,3 +2187,25 @@ pub unsafe fn maybe_enable_freecam(marker: MainThreadMarker) {
         engine::prepend_command(marker, "bxt_freecam 1\n");
     }
 }
+
+fn enable_mouse_look(marker: MainThreadMarker) {
+    sdl::set_relative_mouse_mode(marker, true);
+    client::activate_mouse(marker, true);
+}
+
+fn disable_mouse_look(marker: MainThreadMarker) {
+    sdl::set_relative_mouse_mode(marker, false);
+    client::activate_mouse(marker, false);
+}
+
+fn get_viewangles(marker: MainThreadMarker) -> [f32; 3] {
```

Comment:
Could you make this and the next function unsafe instead of having the unsafe inside? I think it makes more sense. Then in the callbacks, do `get_viewangles: &|| unsafe { get_viewangles(marker) },`